src/HomeKit/HMEnums.cs

```diff
@@ -1136,6 +1140,18 @@ public enum HMSignificantEvent {
 		Sunset,
 	}
 
+	[iOS (13,0), TV (13,0), Watch (6,0), NoMac]
+	public enum HMHomeManagerAuthorizationStatus : long {
```

Comment:
there's no need to use `: long` here - the default 32bits enum has ample space

Reply:
I just kinda followed what was already in the file but yes if It's better to use the default 32bits let's go for it.
e.g. : 
```C#
[iOS (9,0)]
[TV (10,0)]
[Native]
public enum HMCharacteristicValueAirQuality : long {
	Unknown = 0,
	Excellent,
	Good,
	Fair,
	Inferior,
	Poor
}
```

Reply:
I see, but that's not the same case :) 

A `[Native]` enum means it exists, as an `enum` in Objective-C and we **must** use the same size (which is `NSInteger`, so 64 bits open 64 bits architectures). Also special care is needed when used in 32bits (e.g. for p/invoke).

But in this case you are creating a new enum (there's nothing in ObjC to match it) to better represent string constants (more type safe, helping code completion...). Since there's no contract (w/ObjC) you can use the default (smaller) enum size (which is `int` for C#).

If this was a `[Flag]` enum and you could not be sure if more than 32 bits would be needed (in the future) then `[u]long` could make sense too.

Reply:
Wow thanks for the detailed answer 👍